src/motion_planning/scripts/task_runner.py

```diff
@@ -1,44 +1,23 @@
+#!/usr/bin/env python
+
 import rospy
 from task import Task
 from template_tasks import *
-from competition_task import CompetitionTask
 
 class TaskRunner(object):
+    RATE = 30 # Hz
 
-	RATE = 30 # Hz
```

Comment:
Why do you take all of the Task State and put it in another class? This makes it more difficult to trace and I do not see what it adds.

Reply:
1. The stuff that's currently stored as state in a task isn't really task specific. It's more the overall state of the robot. It doesn't make sense to keep duplicating this and it's more intuitive to just store this in a single place.

2. Constantly creating and destroying publishers and subscribers is expensive. They need to be registered with ROS and the unnecessary object creation creates GC pressure.